Feat: Reorganize Wallet PopUp

Villarley opened the pull request:
Added the function sortConnectors and applied logic to display available connectors first, followed by unavailable ones.


# Pull Request type

- Feature

Resolves: #932 

### Summary of Changes

- Added the sortConnectors function to sort connectors based on availability, moving unavailable connectors (e.g., Okx Wallet, Bitget Wallet if not installed) to the end of the list.
- Updated the Navbar component to utilize the sorted connectors for a more intuitive wallet selection experience.
- Verified compatibility between @starknet-react/core and starknetkit.

![feature](https://github.com/user-attachments/assets/9be4ff17-884d-4592-bf1e-03da7f47a4a7)
![testnet](https://github.com/user-attachments/assets/56e10d8d-a91e-4c31-be4e-e51f28ba9923)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced sorting functionality for connectors in the Navbar, enhancing the user experience by prioritizing available connectors.
  
- **Bug Fixes**
	- Improved the handling of connector availability, ensuring a more intuitive display order.

- **Tests**
	- Added a comprehensive test suite for the new sorting functionality to ensure reliability and correct behavior.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Villarley committed:
Feat: Reorganize Wallet PopUp

Added the function sortConnectors and applied logic to display available connectors first, followed by unavailable ones.

fricoben commented on utils/sorts.ts:
```diff
@@ -0,0 +1,24 @@
+import { Connector } from "starknetkit";
```
rename the file `connectors.ts`, we'll reuse it later for connectors utils

fricoben commented on utils/sorts.ts:
```diff
@@ -0,0 +1,24 @@
+      return order.indexOf(a.id) - order.indexOf(b.id);
+    }),
+  ];
+};
```
you need to create a connector.test.js function to test your util and make sur it covers all hedge cases here :)

fricoben reviewed (changes requested):
Test your function :)

Villarley committed:
requested changes

- Updated function import at the navbar file.
- Created connectors.test.js for testing the util function.
- Renamed the file to "connectors.ts"

fricoben reviewed (approved):
lgtm
